Mathlib/Algebra/Polynomial/Degree/IsMonicOfDegree.lean

```diff
@@ -0,0 +1,275 @@
+/-
+Copyright (c) 2025 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Monic
+import Mathlib.Tactic.ComputeDegree
+
+/-!
+# Monic polynomials of given degree
+
+This file defines the predicate `Polynomial.IsMonicOfDegree p n` that states that
+the polynomial `p` is monic and has degree `n` (i.e., `p.natDegree = n`.)
+
+We also provide some basic API.
+-/
+
+namespace Polynomial
+
+/-- This says that `p` has `natDegree` `n` and is monic. -/
+def IsMonicOfDegree {R : Type*} [Semiring R] (p : R[X]) (n : ℕ) : Prop :=
+  p.natDegree = n ∧ p.Monic
+
+lemma IsMonicOfDegree.monic {R : Type*} [Semiring R] {p : R[X]} {n : ℕ}
```

Comment:
Can you please use a `variable {R : Type*} ...` line?

Reply:
And also opening a namespace seems a good idea.

Reply:
I did that now (grouping `[Nontrivial R]` lemmas, except `isMonicOfDegree_iff`, which is needed earlier).